`pants tailor --check ::` fails with `Unknown flag --check on global scope` with new PEX distribution mechanism

**Describe the bug**
In 2.18.0.dev5 and beyond, `pants tailor --check ::` stops working when run under scie-pants 0.9.0 or greater.

This invocation is likely to be widespread, as it is suggested by [docs](https://www.pantsbuild.org/docs/using-pants-in-ci#approach-2-run-over-everything) and [examples](https://github.com/pantsbuild/example-python/blob/751db8be2ed502dba733d8bc1c340064bb9d1995/.github/workflows/pants.yaml#L52). It'd be _really_ good to not break (at least, not without deprecation warnings)

```
Unknown flag --check on global scope
Did you mean --check-only or --tailor-check?
Use `pants help` to get help.
13:19:23.55 [ERROR] Unknown flags --check on global scope


Use --print-stacktrace for more error details and/or -ldebug for more logs. 
See https://www.pantsbuild.org/v2.18/docs/troubleshooting for common issues.
Consider reaching out for help: https://www.pantsbuild.org/v2.18/docs/getting-help
```

Reproducer:

```shell
cd $(mktemp -d)

cat > pants.toml <<EOF
[GLOBAL]
pants_version = "2.18.0.dev5"

backend_packages = ["pants.backend.python"]
EOF

pants tailor --check ::
```

Changing the version to 2.18.0.dev4 passes (no output).


**Pants version**
2.18.0.dev5 to 2.18.0a0 at the time of writing

scie-pants 0.9.0, 0.9.3

**OS**
macOS and Linux

**Additional info**

- I discovered by a "public repo" test run, e.g. https://github.com/pantsbuild/pants/actions/runs/5957288454/job/16159713022
- Changes between those releases: https://github.com/pantsbuild/pants/compare/release_2.18.0.dev4...release_2.18.0.dev5 . ~Nothing leaps out at me. I haven't run a `git bisect` though.~ See first comment: it's the switch to PEX distributions

--- a/CHANGES.md
+++ b/CHANGES.md
@@ -1,5 +1,20 @@
 # Release Notes
 
+## 0.10.0
+
+This release deprecates support for running against an arbitrary Pants commit using
+`PANTS_SHA=abc123... pants ...`. Pants no longer
+publishes the artifacts required for this for new commits, and so this is becoming less and less
+useful. To replace use of `PANTS_SHA`, do one of:
+
+- Use a released version of Pants.
+- Run pants from sources (for example: `PANTS_SOURCE=/path/to/pants-checkout pants ...`).
+- If these are not appropriate, [let us know what you're using it for](https://www.pantsbuild.org/docs/getting-help).
+
+
+This release also fixes scie-pants running commands like `pants tailor --check ...` with Pants
+2.18.0.dev5 and newer (releases that use the new "per-platform" PEX).
+
 ## 0.9.3
 
 This release propagates the version of the `scie-pants` into the invocation of pants, so that Pants

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -6,7 +6,7 @@ members = [
 [package]
 name = "scie-pants"
 description = "Protects your Pants from the elements."
-version = "0.9.3"
+version = "0.10.0"
 edition = "2021"
 authors = [
     "John Sirois <john.sirois@gmail.com>",

--- a/src/main.rs
+++ b/src/main.rs
@@ -8,7 +8,7 @@ use std::path::PathBuf;
 
 use anyhow::{anyhow, bail, Context, Result};
 use build_root::BuildRoot;
-use log::{info, trace};
+use log::{info, trace, warn};
 use logging_timer::{time, timer, Level};
 use uuid::Uuid;
 
@@ -173,11 +173,19 @@ fn get_pants_process() -> Result<Process> {
 
     let env_pants_sha = env_version("PANTS_SHA")?;
     let env_pants_version = env_version("PANTS_VERSION")?;
-    if let (Some(pants_sha), Some(pants_version)) = (&env_pants_sha, &env_pants_version) {
-        bail!(
-            "Both PANTS_SHA={pants_sha} and PANTS_VERSION={pants_version} were set. \
-            Please choose one.",
-        )
+    if let Some(pants_sha) = &env_pants_sha {
+        // when support for PANTS_SHA is fully removed, PANTS_SHA_FIND_LINKS can be removed too
+        warn!(
+            "DEPRECATED: Support for PANTS_SHA={pants_sha} will be removed in a future version of the `pants` launcher. \
+            In addition, the artifacts for PANTS_SHA are no longer published for new commits. Use a released version."
+        );
+
+        if let Some(pants_version) = &env_pants_version {
+            bail!(
+                "Both PANTS_SHA={pants_sha} and PANTS_VERSION={pants_version} were set. \
+                Please choose one.",
+            )
+        }
     }
 
     let pants_version = if let Some(env_version) = env_pants_version {

--- a/src/main.rs
+++ b/src/main.rs
@@ -8,7 +8,7 @@ use std::path::PathBuf;
 
 use anyhow::{anyhow, bail, Context, Result};
 use build_root::BuildRoot;
-use log::{info, trace, warn};
+use log::{info, trace};
 use logging_timer::{time, timer, Level};
 use uuid::Uuid;
 
@@ -175,9 +175,10 @@ fn get_pants_process() -> Result<Process> {
     let env_pants_version = env_version("PANTS_VERSION")?;
     if let Some(pants_sha) = &env_pants_sha {
         // when support for PANTS_SHA is fully removed, PANTS_SHA_FIND_LINKS can be removed too
-        warn!(
-            "DEPRECATED: Support for PANTS_SHA={pants_sha} will be removed in a future version of the `pants` launcher. \
-            In addition, the artifacts for PANTS_SHA are no longer published for new commits. Use a released version."
+        eprintln!(
+            "DEPRECATED: Support for PANTS_SHA=... will be removed in a future version of the `pants` launcher.\
+             \n\nYou've set PANTS_SHA={pants_sha}. The artifacts for PANTS_SHA are no longer published for new commits. \
+             To resolve, use a released version, and/or let us know about what you're doing at <https://www.pantsbuild.org/docs/getting-help>.\n"
         );
 
         if let Some(pants_version) = &env_pants_version {

--- a/src/main.rs
+++ b/src/main.rs
@@ -176,9 +176,16 @@ fn get_pants_process() -> Result<Process> {
     if let Some(pants_sha) = &env_pants_sha {
         // when support for PANTS_SHA is fully removed, PANTS_SHA_FIND_LINKS can be removed too
         eprintln!(
-            "DEPRECATED: Support for PANTS_SHA=... will be removed in a future version of the `pants` launcher.\
-             \n\nYou've set PANTS_SHA={pants_sha}. The artifacts for PANTS_SHA are no longer published for new commits. \
-             To resolve, use a released version, and/or let us know about what you're doing at <https://www.pantsbuild.org/docs/getting-help>.\n"
+          "\
+DEPRECATED: Support for PANTS_SHA=... will be removed in a future version of the `pants` launcher.
+
+The artifacts for PANTS_SHA are no longer published for new commits. This invocation set PANTS_SHA={pants_sha}.
+
+To resolve, do one of:
+- Use a released version of Pants.
+- Run pants from sources (for example: `PANTS_SOURCE=/path/to/pants-checkout pants ...`).
+- If these are not appropriate, let us know what you're using it for: <https://www.pantsbuild.org/docs/getting-help>.
+"
         );
 
         if let Some(pants_version) = &env_pants_version {

--- a/tools/src/scie_pants/configure_pants.py
+++ b/tools/src/scie_pants/configure_pants.py
@@ -125,7 +125,9 @@ def main() -> NoReturn:
     with open(env_file, "a") as fp:
         if resolve_info.find_links:
             print(f"FIND_LINKS={resolve_info.find_links}", file=fp)
-            print(f"PANTS_SHA_FIND_LINKS={resolve_info.pants_find_links_option(version)}", file=fp)
+        # This can be removed once we stop supporting PANTS_SHA:
+        # NB. this is added unconditionally because it gets set as an argument
+        print(f"PANTS_SHA_FIND_LINKS={resolve_info.pants_find_links_option(version)}", file=fp)
         if newly_created_build_root:
             print(f"PANTS_BUILDROOT_OVERRIDE={newly_created_build_root}", file=fp)
         print(f"PANTS_VERSION={version}", file=fp)

--- a/tools/src/scie_pants/pants_version.py
+++ b/tools/src/scie_pants/pants_version.py
@@ -38,10 +38,6 @@ class ResolveInfo:
     find_links: str | None
 
     def pants_find_links_option(self, pants_version_selected: Version) -> str:
-        assert (
-            self.find_links is not None
-        ), "pants_find_links_option shouldn't be called if find_links is None"
-
         # We only want to add the find-links repo for PANTS_SHA invocations so that plugins can
         # resolve Pants the only place it can be found in that case - our ~private
         # binaries.pantsbuild.org S3 find-links bucket.
@@ -51,7 +47,11 @@ class ResolveInfo:
             if self.stable_version in SpecifierSet("<2.14.0", prereleases=True)
             else "find-links"
         )
-        return f"--python-repos-{option_name}={operator}['{self.find_links}']"
+        value = f"'{self.find_links}'" if self.find_links else ''
+
+        # we usually pass a no-op, e.g. --python-repos-find-links=-[], because this is only used for
+        # PANTS_SHA support that is now deprecated and will be removed
+        return f"--python-repos-{option_name}={operator}[{value}]"
 
 
 def determine_find_links(

--- a/tools/src/scie_pants/pants_version.py
+++ b/tools/src/scie_pants/pants_version.py
@@ -47,7 +47,7 @@ class ResolveInfo:
             if self.stable_version in SpecifierSet("<2.14.0", prereleases=True)
             else "find-links"
         )
-        value = f"'{self.find_links}'" if self.find_links else ''
+        value = f"'{self.find_links}'" if self.find_links else ""
 
         # we usually pass a no-op, e.g. --python-repos-find-links=-[], because this is only used for
         # PANTS_SHA support that is now deprecated and will be removed

